src/main/java/com/marklogic/kafka/connect/source/SerializedQueryHandler.java

```diff
@@ -96,6 +115,65 @@ private String buildMaxValueSerializedQuery() throws JsonProcessingException {
         return mapper.writeValueAsString(queryRoot);
     }
 
+    private ObjectNode buildMaxValueQueryConstraintNode() {
+        return mapper.createObjectNode()
+            .put("ns", "op")
+            .put("fn", "select")
+            .set("args", mapper.createArrayNode()
+                .add(mapper.createArrayNode()
+                    .add(mapper.createObjectNode()
+                        .put("ns", "op")
+                        .put("fn", "as")
+                        .set("args", mapper.createArrayNode()
+                            .add("constraint")
+                            .add(mapper.createObjectNode()
+                                .put("ns", "op")
+                                .put("fn", "col")
+                                .set("args", mapper.createArrayNode().add(constraintColumnName))
+                            )
+                        )
+                )
+            ));
+    }
+
+    private ObjectNode buildLimitNode(Integer limit) {
+        return mapper.createObjectNode()
+            .put("ns", "op")
+            .put("fn", "limit")
+            .set("args", mapper.createArrayNode().add(limit));
+    }
+
+    private ObjectNode buildOrderByNode(boolean ascending) {
+        return mapper.createObjectNode()
+            .put("ns", "op")
+            .put("fn", "order-by")
+            .set("args", mapper.createArrayNode()
+                .add(mapper.createObjectNode()
+                    .put("ns", "op")
+                    .put("fn", ascending ? "asc" : "desc")
+                    .set("args", mapper.createArrayNode().add(constraintColumnName))
+            ));
+    }
+
+    private ObjectNode buildConstraintNode(String previousMaxConstraintColumnValue) {
+        return mapper.createObjectNode()
+            .put("ns", "op")
+            .put("fn", "where")
+            .set("args", mapper.createArrayNode()
+                .add(mapper.createObjectNode()
+                    .put("ns", "op")
+                    .put("fn", "gt")
+                    .set("args", mapper.createArrayNode()
+                        .add(mapper.createObjectNode()
+                            .put("ns", "op")
+                            .put("fn", "col")
+                            .set("args", mapper.createArrayNode().add(constraintColumnName))
+                        )
+                        .add(previousMaxConstraintColumnValue)
```

Comment:
I realize now that this has the added benefit of avoiding any issues with concatenating in a value.